doc/design/logging.md

```diff
@@ -0,0 +1,385 @@
+# Logging
+
+## Summary and problem description
+
+Logging is a crucial part of a framework since it informs the developer and
+user of the framework about anomalies and helps with debugging by providing
+contextual information.
+
+Additionally, when integrated into a separate framework, the log messages should
```

Comment:
Not a strong must but here I would introduce a little requirements subsection and list the following 3 sentences as the 3 requirements we have to fulfill.

Then it is crystal clear what we aim for here.